### PR TITLE
Feat: Video page

### PR DESCRIPTION
**Pull request by spraveenofficial**
Added Video Player.
Added API Actions.
Connected Actions to Reducers.
Made Custom Hooks for Every API Calls to manage States.

### CODE REVIEW

File: frontend/src/Actions/auth.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  USER_REGISTER_SUCCESS,
+  USER_REGISTER_FAILURE,
+} from "../Constants/auth";
+import axios from "axios";
+import baseUrl from "../Utils/baseurl";
+export const loginUser = async (payload, dispatch) => {
+  try {
+    dispatch({ type: USER_LOGIN_REQUEST });
+    const { data } = await axios({
+      method: "POST",
```
Comment (tanishk15sharma) [inline frontend/src/Actions/auth.js:18]: this approach to `POST` data in an obj is good

**Review by tanishk15sharma — APPROVED** [review]
amazing work !!

File: frontend/src/Components/Player/index.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    ],
+  };
+  useEffect(() => {
+    const vjsPlayer = videojs(videoRef.current);
+    if (previewUrl) {
+      vjsPlayer.src({ type: "video/mp4", src: previewUrl });
+    }
+
+    // vjsPlayer.on("ended", () => {
+    //   client(`${process.env.REACT_APP_BE}/videos/${videoId}/view`);
```
Comment (Prabhav-Sharma) [inline frontend/src/Components/Player/index.jsx:27]: Remove unuseful commented code

**Review by Prabhav-Sharma — APPROVED** [review]
Looks good, try to make smaller PRs

**Review by sharathnair9999 — APPROVED** [review]

**Pull request merged**